### PR TITLE
Somil55/non visitor based mincut

### PR DESCRIPTION
**Pull request by somil55**
```
julia> g = Graph(5000,100000)
{5000, 100000} undirected simple Int64 graph

julia> is_connected(g)
true

julia> using BenchmarkTools

julia> @benchmark mincut(g)
BenchmarkTools.Trial: 
  memory estimate:  20.49 MiB
  allocs estimate:  1271886
  --------------
  minimum time:     67.345 ms (0.00% GC)
  median time:      69.124 ms (1.39% GC)
  mean time:        81.092 ms (1.16% GC)
  maximum time:     131.464 ms (0.97% GC)
  --------------
  samples:          63
  evals/sample:     1

julia> @benchmark mincut_new(g)
BenchmarkTools.Trial: 
  memory estimate:  666.34 KiB
  allocs estimate:  46
  --------------
  minimum time:     22.481 ms (0.00% GC)
  median time:      28.689 ms (0.00% GC)
  mean time:        26.937 ms (0.05% GC)
  maximum time:     34.832 ms (0.00% GC)
  --------------
  samples:          186
  evals/sample:     1

julia> mincut_new(g) == mincut(g)
true

### CODE REVIEW

**Comment by somil55** [comment]
```
julia> @benchmark mincut(Graph(5000,100000))
BenchmarkTools.Trial: 
  memory estimate:  41.23 MiB
  allocs estimate:  1487744
  --------------
  minimum time:     109.731 ms (1.47% GC)
  median time:      149.045 ms (3.31% GC)
  mean time:        166.509 ms (3.02% GC)
  maximum time:     272.934 ms (2.24% GC)
  --------------
  samples:          31
  evals/sample:     1

julia> @benchmark mincut_new(Graph(5000,100000))
BenchmarkTools.Trial: 
  memory estimate:  21.68 MiB
  allocs estimate:  230334
  --------------
  minimum time:     62.625 ms (2.81% GC)
  median time:      69.261 ms (3.33% GC)
  mean time:        79.507 ms (3.14% GC)
  maximum time:     157.153 ms (2.77% GC)
  --------------
  samples:          64
  evals/sample:     1

```

File: src/traversals/maxadjvisit.jl [hunk 1]
```diff
@@ -161,6 +161,65 @@ function mincut(
...
+    distmx::AbstractMatrix{T}=weights(g)
+) where T
+
+    U = eltype(g)
+    colormap = zeros(Int8, nv(g))   ## 0 if unseen, 1 if processing and 2 if seen and closed
+    parities = falses(nv(g))
+    bestweight = typemax(T)
+    cutweight = zero(T)
+    visited = zero(Int)             ## number of vertices visited
+    # vertices = Vector{U}()        ## not sure if this extra info is necessary to be stored
```
Comment (somil55) [inline src/traversals/maxadjvisit.jl]: @sbromberger do you need this information?

Comment (sbromberger) [reply src/traversals/maxadjvisit.jl]: I don't think so. Let's leave it out.

**Comment by sbromberger** [comment]
Those are some pretty impressive benchmarks!

**Comment by somil55** [comment]
@sbromberger can u fix this ?

**Comment by sbromberger** [comment]
@somil55 - sorry for the delay. What you need to do is a "git merge master" and then resolve the conflicts locally, then commit those changes and push again.

**Pull request merged**